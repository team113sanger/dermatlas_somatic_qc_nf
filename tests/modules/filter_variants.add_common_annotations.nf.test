nextflow_process {

    name "Test Process ADD_COMMON_ANNOTATIONS"
    script "modules/filter_variants.nf"
    process "ADD_COMMON_ANNOTATIONS"

    test("Should run without failures") {

        when {
        params {
            load("$baseDir/tests/testdata/test_params.json")
        }
        process {
                """
                input[0] = tuple([sample_id: "example", caller: "pindel"], 
                file("$baseDir/tests/testdata/example.filt.vcf.gz"), file("$baseDir/tests/testdata/example.filt.vcf.gz.tbi"))
                input[1] = file("${params.dbsnp_variants}")
                input[2] = file("${params.dbsnp_header}")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
