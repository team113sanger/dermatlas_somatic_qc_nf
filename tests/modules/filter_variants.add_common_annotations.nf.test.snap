{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sample_id": "example",
                            "caller": "pindel",
                            "filename": "TBC",
                            "vcfoutdir": "here"
                        },
                        "TBC.filt.snpflagged.vcf.gz:md5,95cded79cbd4be7e5d83ab90aeee55fc",
                        "TBC.filt.snpflagged.vcf.gz.tbi:md5,adac6ac92185f5781f2618f177d4613f"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-09-18T11:05:27.860332"
    }
}